homeassistant/components/evohome.py

```diff
@@ -71,75 +70,110 @@ def setup(hass, config):
+                "Wait a while (say 10 minutes) and restart HA."
+            )
+
+        else:
+            raise  # we dont expect/handle any other HTTPErrors
+
+        return False  # unable to continue
+
+    finally:  # Redact username, password as no longer needed
         evo_data['params'][CONF_USERNAME] = 'REDACTED'
         evo_data['params'][CONF_PASSWORD] = 'REDACTED'
 
     evo_data['client'] = client
 
-    # Redact any installation data we'll never need.
-    if client.installation_info[0]['locationInfo']['locationId'] != 'REDACTED':
-        for loc in client.installation_info:
-            loc['locationInfo']['streetAddress'] = 'REDACTED'
-            loc['locationInfo']['city'] = 'REDACTED'
-            loc['locationInfo']['locationOwner'] = 'REDACTED'
-            loc[GWS][0]['gatewayInfo'] = 'REDACTED'
+    # Redact any installation data we'll never need
+    for loc in client.installation_info:
+        loc['locationInfo']['locationId'] != 'REDACTED'
+        loc['locationInfo']['locationOwner'] = 'REDACTED'
+        loc['locationInfo']['streetAddress'] = 'REDACTED'
+        loc['locationInfo']['city'] = 'REDACTED'
+        loc[GWS][0]['gatewayInfo'] = 'REDACTED'
 
-    # Pull down the installation configuration.
+    # Pull down the installation configuration
     loc_idx = evo_data['params'][CONF_LOCATION_IDX]
 
     try:
         evo_data['config'] = client.installation_info[loc_idx]
 
     except IndexError:
         _LOGGER.warning(
-            "setup(): Parameter '%s' = %s , is outside its range (0-%s)",
+            "setup(): Parameter '%s'=%s , is outside its range (0-%s)",
             CONF_LOCATION_IDX,
             loc_idx,
             len(client.installation_info) - 1
         )
 
         return False  # unable to continue
 
-    evo_data['status'] = {}
-
     if _LOGGER.isEnabledFor(logging.DEBUG):
         tmp_loc = dict(evo_data['config'])
         tmp_loc['locationInfo']['postcode'] = 'REDACTED'
-        tmp_tcs = tmp_loc[GWS][0][TCS][0]
-        if 'zones' in tmp_tcs:
-            tmp_tcs['zones'] = '...'
-        if 'dhw' in tmp_tcs:
-            tmp_tcs['dhw'] = '...'
+        if 'dhw' in tmp_loc[GWS][0][TCS][0]:  # if this location has DHW...
+            tmp_loc[GWS][0][TCS][0]['dhw'] = '...'
+
+        _LOGGER.debug("setup(): evo_data['config']=%s", tmp_loc)
+
+    # evohomeclient has exposed no means of accessing non-default location
+    # (i.e. loc_idx > 0) other than using a protected member, such as below
+    tcs_obj_ref = client.locations[loc_idx]._gateways[0]._control_systems[0]    # noqa E501; pylint: disable=protected-access
+
+    _LOGGER.debug(
+        "setup(): Found Controller, id=%s, name=%s (location_idx=%s)",
+        tcs_obj_ref.systemId + " [" + tcs_obj_ref.modelType + "]",
+        tcs_obj_ref.location.name,
+        loc_idx
+    )
+
+    evo_data['parent'] = EvoController(evo_data, client, tcs_obj_ref)
```

Comment:
Platform entities are not allowed to be created or accessed outside their platforms.